src/components/Image.tsx

```diff
@@ -0,0 +1,134 @@
+import * as React from "react";
+import { ComponentConfig, Fields } from "@measured/puck";
+import { Image, ImageProps } from "@yext/pages-components";
+import {
+  EntityField,
+  YextEntityField,
+  resolveYextEntityField,
+  YextEntityFieldSelector,
+} from "../index.ts";
+import { cn } from "../internal/utils/cn.ts";
+import { cva, type VariantProps } from "class-variance-authority";
+import { useDocument } from "../hooks/index.ts";
+
+const imageWrapperVariants = cva("", {
+  variants: {
+    size: {
+      small: "max-w-xs",
+      medium: "max-w-md",
+      large: "max-w-xl",
+      full: "w-full",
+    },
+    aspectRatio: {
+      auto: "aspect-auto",
+      square: "aspect-square",
+      video: "aspect-video",
+      portrait: "aspect-[3/4]",
+    },
+    rounded: {
+      none: "",
+      small: "rounded-sm",
+      medium: "rounded-md",
+      large: "rounded-lg",
+      full: "rounded-full",
+    },
+  },
+  defaultVariants: {
+    size: "medium",
+    aspectRatio: "auto",
+    rounded: null,
+  },
+});
+
+export interface ImageWrapperProps
+  extends VariantProps<typeof imageWrapperVariants> {
+  image: YextEntityField;
+}
+
+const imageWrapperFields: Fields<ImageWrapperProps> = {
+  image: YextEntityFieldSelector<any>({
+    label: "Image",
+    filter: {
+      types: ["type.image"],
+    },
+  }),
+  size: {
+    type: "select",
+    label: "Size",
+    options: [
+      { label: "Small", value: "small" },
+      { label: "Medium", value: "medium" },
+      { label: "Large", value: "large" },
+      { label: "Full Width", value: "full" },
+    ],
+  },
+  aspectRatio: {
+    type: "select",
+    label: "Aspect Ratio",
+    options: [
+      { label: "Auto", value: "auto" },
+      { label: "Square", value: "square" },
+      { label: "Video (16:9)", value: "video" },
+      { label: "Portrait (3:4)", value: "portrait" },
+    ],
+  },
+  rounded: {
+    type: "select",
+    label: "Rounded Corners",
+    options: [
+      { label: "None", value: "none" },
+      { label: "Small", value: "small" },
+      { label: "Medium", value: "medium" },
+      { label: "Large", value: "large" },
+      { label: "Full", value: "full" },
+    ],
+  },
+};
+
+const ImageWrapper: React.FC<ImageWrapperProps> = ({
+  image: imageField,
+  size,
+  aspectRatio,
+  rounded,
+}) => {
+  const document = useDocument();
+  const resolvedImage = resolveYextEntityField<ImageProps["image"]>(
+    document,
+    imageField
+  );
+
+  console.log(resolvedImage);
+
+  if (!resolvedImage) {
+    return null;
+  }
+
+  return (
+    <EntityField displayName="Image" fieldId={imageField.field}>
+      <div
+        className={cn(
+          imageWrapperVariants({ size, aspectRatio }),
+          rounded && "overflow-hidden rounded-lg"
+        )}
+      >
+        <Image image={resolvedImage} className="w-full h-full object-cover" />
```

Comment:
I think we might want to allow someone to set up a constant value which would allow them to give an http url. Probably other fields required by Image as well.

Reply:
To do this I think we will need to support type.image as a constant value type

https://github.com/yext/visual-editor/blob/main/src/components/editor/YextEntityFieldSelector.tsx#L24

https://github.com/yext/visual-editor/pull/102/files#diff-2650600b809d4593f7d2a1efa547ef3e03a4583c3287603546655298cb4d8828R52

Reply:
Going to add constant field support for images as a separate PR so we can close this out